### ISSUE TITLE
NFC features for LNURL-w

### ISSUE THREAD

**Issue by benjamin-wilson** [ZeusLN/zeus#990]
### Describe the feature

The goal of this feature is to emulate being your own credit card provider. Here is one way you could implement it:

On the main page of zeus, under the Lightning & On-chain sections there could be a similarly styled button called 'NFC', 'Cards', 'NFC-Cards', 'LNURL Withdraw', 'NFC LNURL-w' or something similar. 

Clicking this button would have 'send' and 'mange' settings.

Clicking 'send' would open a dialog where you could enter the max amount for the LNURL-w and then another button to use host-based card emulation to send the LNURL-w over NFC (disposable).
https://developer.android.com/guide/topics/connectivity/nfc/hce

Clicking 'mange' would allow you to create LNURL-w's and write them to a NFC card. This section would also allow you to read existing NFC cards. You could also keep track of all the cards and their balances. (permanent)
https://docs.lnpay.co/api/lnurl-withdraw/how-does-lnurl-withdraw-work
https://twitter.com/BitcoinQ_A/status/1527286484385992707

**Comment by Bosch-0** [comment]
Could also achieve a lot of this more eloquently and natively using Offers. Great suggestion though will put it on our backlog to look into. Doing stuff with NFC cards is something we've discussed doing for awhile

**Comment by benjamin-wilson** [comment]
I agree unfortunately LND doesn't support BOLT 12 yet and doesn't look like it'll be implemented soon. 
https://github.com/lightningnetwork/lnd/issues/5594#issuecomment-1042314431

**Comment by kaloudis** [comment]
Thanks for the suggestion. Revisiting this one, most of the tap-to-pay functionality described here has since shipped:

- NFC reading is supported across the payment surfaces: you can scan LNURL-withdraw cards from the Send and Receive views (ZEUS generates the invoice and submits it to the withdraw callback for you), take Boltcard taps in POS mode, and tapping a `lightning:` tag launches the app on Android.
- Phone-to-phone NFC payments work via host card emulation on Android: the receiving device can broadcast an invoice, on-chain address, BOLT 12 offer, lightning address, or Cashu token as an emulated NFC tag from any QR view, and the sending device picks it up with the NFC scan button on the Send view. (iOS doesn't allow third-party HCE, so broadcasting is Android-only.)

What remains from the original request is creating LNURL-withdraws and writing them to physical cards. An LNURL-withdraw requires an always-on server holding spendable funds, which a self-custodial mobile wallet can't provide on its own, and we aren't going to run that as a custodial service on users' behalf. Closing this out, since the parts we can do self-custodially are done.
